src/Agent.Worker/Release/ArtifactProvider.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+using Agent.Worker.Release.Artifacts.Definition;
+
+using Microsoft.VisualStudio.Agent.Worker.Release.Artifacts;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.VisualStudio.Services.Agent.Worker.Release.Artifacts;
+using Microsoft.VisualStudio.Services.ReleaseManagement.WebApi.Contracts;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.Release
```

Comment:
It looks like a more intuitive design here would be:
IArtifactProviderManager (abstract factory for returning the correct provider)
IArtifactProvider (same interface you have here)

Then you could delete all of the duplicated interfaces: IBuildArtifact, IFileShareArtifact, IJenkinsArtifact